src/sentry/utils/strings.py

```diff
@@ -41,9 +42,13 @@ def unicode_escape_recovery_handler(err):
 
 def unescape_string(value):
     """Unescapes a backslash escaped string."""
-    return value.encode("ascii", "backslashreplace").decode(
-        "unicode-escape", "unicode-escape-recovery"
-    )
+    with warnings.catch_warnings():
```

Comment:
I don't think we can do this, catch_warnings looks like it applies to the entire process rather than the current thread

not sure if there's any easier way

do you have a link to the sentry issue?

Reply:
https://sentry.io/organizations/sentry/issues/2476704750/?project=1

Ah yeah, that's unfortunate :\ 

I'll opt for fixing the files instead of `catch_warnings`.